win-build.ps1

```diff
@@ -0,0 +1,206 @@
+        Write-Host "Making sure Chocolatey is current."
+        choco upgrade chocolatey --confirm | Out-Null
+    }
+
+    # We need the native library dependencies for `hab`
+    # Until we have habitat packages on Windows, there is
+    # a chocolatey package hosted in MyGet with the native 
+    # dependencies built.
+    if ((choco list habitat_native_dependencies --local-only) -match '^1 packages installed\.$') {
+        choco upgrade habitat_native_dependencies --confirm -s https://www.myget.org/F/habitat/api/v2  --allowemptychecksums
+    } 
+    else {
+        choco install habitat_native_dependencies --confirm -s https://www.myget.org/F/habitat/api/v2  --allowemptychecksums
+    }
+
+    # set a few reference variables for later
+    $ChocolateyHabitatLibDir = "$env:ChocolateyInstall\lib\habitat_native_dependencies\builds\lib"
+    $ChocolateyHabitatIncludeDir = "$env:ChocolateyInstall\lib\habitat_native_dependencies\builds\include"
+    $ChocolateyHabitatBinDir = "$env:ChocolateyInstall\lib\habitat_native_dependencies\builds\bin"
+
+    if (-not (Test-AppVeyor)) {
+        # We need the Visual C 2013 Runtime for the Win32 ABI Rust
+        choco install 'vcredist2013' --confirm --allowemptychecksum
+
+        # We need the Visual C++ tools to build Rust crates (provides a compiler and linker) 
+        choco install 'visualcppbuildtools' --version '14.0.25123' --confirm --allowemptychecksum
+
+        choco install 7zip --version '16.02.0.20160811' --confirm
+    }
+
+    # Install Rust Nightly (since there aren't MSVC nightly cargo builds)
+    if (get-command -Name rustup.exe -ErrorAction SilentlyContinue) {
+        rustup install stable-x86_64-pc-windows-msvc
+        $cargo = 'rustup run stable-x86_64-pc-windows-msvc cargo'
+    }
+    else {
+        $env:PATH = New-PathString -StartingPath $env:PATH -Path "C:\Program Files\Rust stable MSVC 1.12\bin"
+        if (-not (get-command rustc -ErrorAction SilentlyContinue)) {
+            write-host "installing rust"
+            Invoke-WebRequest -UseBasicParsing -Uri 'https://static.rust-lang.org/dist/rust-1.12.0-x86_64-pc-windows-msvc.msi' -OutFile "$env:TEMP/rust-12-stable.msi"
+            start-process -filepath MSIExec.exe -argumentlist "/qn", "/i", "$env:TEMP\rust-12-stable.msi" -Wait
+            $env:PATH = New-PathString -StartingPath $env:PATH -Path "C:\Program Files\Rust stable MSVC 1.12\bin"
+            while (-not (get-command cargo -ErrorAction SilentlyContinue)) {
+                Write-Warning "`tWaiting for `cargo` to be available."
+                start-sleep -Seconds 1
+            }
+        }
+        else {
+            # TODO: version checking logic and upgrades
+        }
+    }
+}
+
+function Invoke-Build([string]$Path, [switch]$Clean, [switch]$Release) {
+    $Path = Resolve-Path $Path
+
+    Push-Location "$Path"
+    if($Clean) { invoke-expression "cargo clean" }
+    Invoke-Expression "cargo build $(if ($Release) { '--release' })" -ErrorAction Stop
```

Comment:
This may be a problem in `rustup` environments if `stable-msvc` isn't the default or override rust environment.